mmdeploy/core/rewriters/function_rewriter.py

```diff
@@ -72,7 +73,16 @@ def _set_func(origin_func_path: str,
             rewrite_func,
             ignore_refs=ignore_refs,
             ignore_keys=ignore_keys)
-    exec(f'{origin_func_path} = rewrite_func')
+
+    is_static_method = False
+    if method_class:
+        origin_type = inspect.getattr_static(module_or_class, split_path[-1])
+        is_static_method = isinstance(origin_type, staticmethod)
+
+    if is_static_method:
+        exec(f'{origin_func_path} = staticmethod(rewrite_func)')
+    else:
+        exec(f'{origin_func_path} = rewrite_func')
```

Comment:
why do we make difference between static methods and other methods? Removing `self` for beginners might be confusing compared with original way of rewriting.

Reply:
1. static method does not have self 
https://github.com/open-mmlab/mmdetection3d/blob/47285b3f1e9dba358e98fcd12e523cfd0769c876/mmdet3d/core/bbox/coders/fcos3d_bbox_coder.py#L103
2. static method need special treatment, or it won't recover after context exit.

Reply:
I see.